packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -508,29 +505,28 @@ class SyncWatcher extends BaseWatcher {
 
   async handleTransferRootBondedEvent (event: TransferRootBondedEvent) {
     const { transactionHash, blockNumber } = event
-    const { root, amount } = event.args
-    const logger = this.logger.create({ root })
+    const { root: transferRootHash, amount } = event.args
+    const transferRootId = this.bridge.getTransferRootId(
+      transferRootHash,
+      amount
+    )
+
+    const logger = this.logger.create({ root: transferRootId })
     logger.debug('handling TransferRootBonded event')
 
     try {
-      const bondTransferRootId = this.bridge.getTransferRootId(
-        root,
-        amount
-      )
-
-      logger.debug(`transferRootHash from event: ${root}`)
+      logger.debug(`transferRootHash from event: ${transferRootHash}`)
       logger.debug(`event transactionHash: ${transactionHash}`)
       logger.debug(`event blockNumber: ${blockNumber}`)
       logger.debug(`bondAmount: ${this.bridge.formatUnits(amount)}`)
-      logger.debug(`bondTransferRootId: ${bondTransferRootId}`)
+      logger.debug(`transferRootId: ${transferRootId}`)
 
-      await this.db.transferRoots.update(root, {
-        transferRootHash: root,
+      await this.db.transferRoots.update(transferRootId, {
+        transferRootHash,
         bonded: true,
         bondTotalAmount: amount,
```

Comment:
I believe it now makes sense to completely get rid of the concept of `bondTotalAmount`, since only one `totalAmount` can exist per `transferId` now. Based on what I'm seeing (and after one comment below is addressed), I think it should be super easy to completely get rid of the concept of `bondTotalAmount`.

**This DB entry does need to now add `totalAmount: amount`**, however, because we need that available everywhere. See one of the bullets on the top level comment on the PR for my thoughts there.

Reply:
I believe you are correct. The concept of `bondTotalAmount` is removed in https://github.com/hop-protocol/hop/pull/318/commits/43a9da3c4f59b6788f81be76eb08d43b82098aca as well as using `transferRootId` only in challenge watcher

Reply:
This change should also add the following to the DB during this `update`, since it is possible that this is the first instance of a `transferRoot` entry being added to the DB:

```
      await this.db.transferRoots.update(transferRootId, {
        ...
        totalAmount: amount,
        ...
      })
```

Reply:
Great catch, added in https://github.com/hop-protocol/hop/pull/318/commits/75db07adbd7005fadbda27102c80d24358b9789b